packages/bsky/src/services/feed/util.ts

```diff
@@ -0,0 +1,111 @@
+import { sql } from 'kysely'
+import { AtUri } from '@atproto/syntax'
+import {
+  Record as PostRecord,
+  ReplyRef,
+} from '../../lexicon/types/app/bsky/feed/post'
+import {
+  Record as GateRecord,
+  isFollowingRule,
+  isListRule,
+  isMentionRule,
+} from '../../lexicon/types/app/bsky/feed/gate'
+import { isMention } from '../../lexicon/types/app/bsky/richtext/facet'
+import { valuesList } from '../../db/util'
+import DatabaseSchema from '../../db/database-schema'
+import { ids } from '../../lexicon/lexicons'
+
+export const checkInvalidReplyParent = (
+  reply: ReplyRef,
+  parent: { isInvalidReply: boolean | null; record: PostRecord },
+) => {
+  const replyRoot = reply.root.uri
+  const replyParent = reply.parent.uri
+  // if parent is not a valid reply, transitively this is not a valid one either
+  if (parent.isInvalidReply) {
+    return true
+  }
+  // replying to root post: ensure the root looks correct
+  if (replyParent === replyRoot) {
+    return !!parent.record.reply
+  }
+  // replying to a reply: ensure the parent is a reply for the same root post
+  return parent.record.reply?.root.uri !== replyRoot
+}
+
+export const checkInvalidInteractions = async (
```

Comment:
nit, but this is a single interaction

maybe `isInvalidInteraction`?

Reply:
similarly to `checkInvalidReplyParent` the "check" suggests to me that this returns more info than just a boolean

sorry this is such a nit 😅 i just had to check the fn impl to figure out what these were doing cause the name wasn't immediatley clear to me